apps/cowswap-frontend/src/modules/hooksStore/pure/AddCustomHookForm/CustomDappLoader/index.tsx

```diff
@@ -25,27 +45,70 @@ export function ExternalDappLoader({
 }: ExternalDappLoaderProps) {
   const { chainId } = useWalletInfo()
 
-  useEffect(() => {
-    let isRequestRelevant = true
+  const fetchManifest = useCallback(
+    async (url: string) => {
+      if (!url) return
+
+      setLoading(true)
+
+      try {
+        // Basic URL validation first
+        let urlObject: URL
```

Comment:
`urlObject` should be const inside of try {}

Reply:
Addressed.